packages/client/components/PokerDimensionValueControl.tsx

```diff
@@ -76,9 +78,42 @@ interface Props {
   stage: PokerDimensionValueControl_stage
 }
 
+const useHandleFinalScoreError = ({error, serviceField, stageId}) => {
+  const atmosphere = useAtmosphere()
+  const {closePortal, openPortal, modalPortal} = useModal()
+
+  useEffect(() => {
+    const nextError = error?.message ?? ''
+
+    if (
+      nextError &&
+      nextError.includes(
+        `Update failed! In Jira, add the field "${serviceField.name}" to the Issue screen.`
+      )
+    ) {
+      openPortal()
```

Comment:
With the current solution, if the same error happens after attempting to add Jira field it won't show the modal because `useEffect` will not fire (error string, the dependency will be the same).

If you're asking in general about _useEffect_ here vs callback, the _useEffect_ might not be the best choice if `openPortal` is mutating the dom (and now that I see, it is) and the user might see some flickering as it doesn't block the browser painting. In this case, `useLayoutEffect` might be a better choice. I'm not an expert there so correct me If I'm wrong, please.

If we let the `PokerDimensionValueControl` component handle the errors as I described in the comment below the best way would be to put it in `onCompleted` handler of `PokerSetFinalScoreMutation` like that:
```
    PokerSetFinalScoreMutation(
      atmosphere,
      {finalScore: pendingScore, meetingId, stageId},
      {
        onError,
        onCompleted: (res, errors) => {
          onCompleted(res, errors)

          const mutationError = res.pokerSetFinalScore?.error?.message ?? ''
          if (
            mutationError.includes(
              `Update failed! In Jira, add the field "${serviceField.name}" to the Issue screen.`
            )
          ) {
          }
        }
      }
    )
```

Reply:
beautifully explained!

Functionally, there's no real difference between putting the logic in a hook vs. the callback. My reason for preferring the `openPortal` logic inside the callback is because it's closer to the triggering event, which makes it easier to debug. Also fun, while it doesn't much matter here, it could save us a re-render because it's setting the error & opening the portal in the same call. 

`useLayoutEffect` isn't necessary because the modal is going to take over the entire screen, so we don't need to do any DOM measurements, but you're correct if we were going to do something that was going to both move an element & pop a modal based off the element position.

Reply:
@mattkrick for now I've left it in the same place. I didn't like checking 
```
mutationError.includes(
              `Update failed! In Jira, add the field "${serviceField.name}" to the Issue screen.`
            )
```
two times and I couldn't think of a better solution. if you'd rather do it differently let me know, happy to improve!

Reply:
i'll PR this PR with an alternate